test/fs/test_mount.c

```diff
@@ -26,21 +26,49 @@ int main() {
     contents = FS.readFile('/working/waka.txt', { encoding: 'utf8' });
     assert(contents === 'az');
 
-    // mount to a missing directory
+#if !defined(WASMFS)
+    // the legacy API requires a mount directory to exist, while WasmFS will create the directory.
     try {
       FS.mount(MEMFS, {}, '/missing');
     } catch (e) {
       ex = e;
     }
-    assert(ex instanceof FS.ErrnoError && ex.errno === 44); // ENOENT
+    assert(ex.name === 'ErrnoError' && ex.errno === 44); // ENOENT
+#endif
+
+#if WASMFS
+    // WasmFS will throw an error if a directory to mount to is not empty, while the legacy API will not.
+    try {
+      FS.mkdir("/test");
+      FS.writeFile("/test/hi.txt", "abc");
+      FS.mount(MEMFS, {}, '/test');
+    } catch (e) {
+      ex = e;
+    }
+    assert(ex.name === 'ErrnoError' && ex.errno === 55); // ENOTEMPTY
+#endif
 
     // mount to an existing mountpoint
     try {
       FS.mount(MEMFS, {}, '/working');
     } catch (e) {
       ex = e;
     }
-    assert(ex instanceof FS.ErrnoError && ex.errno === 10); // EBUSY
+#if WASMFS
+    // WasmFS returns ENOTEMPTY instead of EBUSY in exchange for a simpler implementation.
+    assert(ex.name === 'ErrnoError' && ex.errno === 55); // ENOTEMPTY
+#else
+    assert(ex.name === 'ErrnoError' && ex.errno === 10); // EBUSY
+#endif
+
+    // attempt to unmount a non-mountpoint directory inside a mountpoint
+    FS.mkdir('/working/unmountable');
+    try {
+      FS.unmount('/working/unmountable');
+    } catch (e) {
+      ex = e;
+    }
+    assert(ex.name === 'ErrnoError' && ex.errno === 28); // EINVAL
```

Comment:
A paranoid part of me is worried about us not doing `ex = null` after each assert. Otherwise if we look for the same error code twice in a row, the second time could not throw but still pass thanks to the existing value...